lib/src/cards/blog_card.dart

```diff
@@ -0,0 +1,190 @@
+  /// of type `VoidCallback`. This variable is required and will be used as the callback function when
+  /// the blog card is tapped. A `VoidCallback` is a function that takes no arguments and returns no
+  /// value.
+  final VoidCallback onBlogCardTap;
+
+  /// The line `final double? cardWidth;` is declaring a final variable named `cardWidth` of type
+  /// `double?`. The `?` indicates that the variable can be nullable, meaning it can have a value of
+  /// `null`. This variable is optional and can be used to specify the width of the blog card. If no
+  /// value is provided, the default width of 350 will be used.
+  final double? cardWidth;
+
+  /// The line `final VoidCallback? onMoreTap;` is declaring a final variable named `onMoreTap` of type
+  /// `VoidCallback?`. The `?` indicates that the variable can be nullable, meaning it can have a value
+  /// of `null`. This variable is optional and can be used as the callback function when the "More"
+  /// button on the blog card is tapped. If no value is provided, the "More" button will not be
+  /// displayed.
+  final VoidCallback? onMoreTap;
+
+  /// The line `final ButtonStyle? morebtnStyle;` is declaring a final variable named `morebtnStyle` of
+  /// type `ButtonStyle?`. The `?` indicates that the variable can be nullable, meaning it can have a
+  /// value of `null`. This variable is optional and can be used to specify the style of the "More"
+  /// button on the blog card. If no value is provided, the default button style will be used.
+  final ButtonStyle? morebtnStyle;
+
+  /// The line `final String imagelink;` is declaring a final variable named `imagelink` of type
+  /// `String`. This variable is required and will be used as the link to the image that will be
+  /// displayed on the blog card.
+  final String imagelink;
+
+  /// The line `final List<String> circularAvatarImages;` is declaring a final variable named
+  /// `circularAvatarImages` of type `List<String>`. This variable is required and will be used to
+  /// provide a list of image links for circular avatars that will be displayed on the blog card. Each
+  /// image link in the list should be of type `String`.
+  final List<String> circularAvatarImages;
+
+  /// The line `final TextStyle descriptionStyle, titleStyle, dateStyle;` is declaring three final
+  /// variables named `descriptionStyle`, `titleStyle`, and `dateStyle` of type `TextStyle`. These
+  /// variables are optional and can be used to specify the style of the description, title, and date
+  /// text in the blog card. If no value is provided, the default text styles will be used.
+  final TextStyle descriptionStyle, titleStyle, dateStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: onBlogCardTap,
+      child: Material(
+        borderRadius: const BorderRadius.all(Radius.circular(16)),
+        elevation: 5,
+        child: SizedBox(
+          width: cardWidth ?? 350,
```

Comment:
Make it dynamic, see other widgets for reference.

Like you can ask for a width factor from user then multiply that width factor with the `getScreenWidth(context)` and provide a default `widthFactor` like `0.7 or 0.8`

Reply:
Sure , I'll make the necessary changes